util.sh missing when trying to run quickstart instructions

<!-- This form is for bug reports and feature requests ONLY! 

If you're looking for help check [Stack Overflow](https://stackoverflow.com/questions/tagged/kubernetes) and the [troubleshooting guide](https://kubernetes.io/docs/tasks/debug-application-cluster/troubleshooting/).
-->

/kind bug

**What happened**:

Followed the quick start guide here https://github.com/kubernetes/kubernetes#to-start-developing-kubernetes

**What you expected to happen**:

make completed sucessfully

**How to reproduce it (as minimally and precisely as possible)**:
```
zapf(~) % go get -d k8s.io/kubernetes                                                                                 
package k8s.io/kubernetes: no Go files in /Users/dfc/src/k8s.io/kubernetes                                            
zapf(~) % cd $GOPATH/src/k8s.io/kubernetes                                                                            
zapf(~/src/k8s.io/kubernetes) % make                       
hack/make-rules/../../hack/lib/init.sh: line 39: /Users/dfc/src/k8s.io/kubernetes                                     
/Users/dfc/src/k8s.io/kubernetes/hack/lib/util.sh: No such file or directory                                          
make[1]: *** [_output/bin/deepcopy-gen] Error 1            
make: *** [generated_files] Error 2      
```
**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): n/a
- Cloud provider or hardware configuration**: n/a
- OS (e.g. from /etc/os-release): macOS Sierra(10.12)

--- a/hack/lib/init.sh
+++ b/hack/lib/init.sh
@@ -18,6 +18,10 @@ set -o errexit
 set -o nounset
 set -o pipefail
 
+# Unset CDPATH so that path interpolation can work correctly
+# https://github.com/kubernetes/kubernetes/issues/52255
+unset CDPATH
+
 # The root of the build/dist directory
 KUBE_ROOT="$(cd "$(dirname "${BASH_SOURCE}")/../.." && pwd -P)"
 

